v1/ast/annotations.go

```diff
@@ -752,10 +752,7 @@ func (c *CompileAnnotation) Compare(other *CompileAnnotation) int {
 		return -1
 	}
 
-	if cmp := slices.CompareFunc(c.Unknowns, other.Unknowns,
-		func(x, y Ref) int {
-			return x.Compare(y)
-		}); cmp != 0 {
+	if cmp := slices.CompareFunc(c.Unknowns, other.Unknowns, RefCompare); cmp != 0 {
```

Comment:
```suggestion
	if cmp := slices.CompareFunc(c.Unknowns, other.Unknowns, Ref.Compare); cmp != 0 {
```
I had expected this. 😄

Reply:
Yeah, that would be neat! But sadly it doesn’t work :( Something about the method getting passed a Value rather than a Ref, and the compiler therefore rejecting the  reference as not satisfying the type constraint.